lib/engine-universal/src/engine.rs

```diff
@@ -498,7 +501,49 @@ impl Engine for UniversalEngine {
         binary: &[u8],
         tunables: &dyn Tunables,
     ) -> Result<Box<dyn wasmer_engine::Executable>, CompileError> {
-        self.compile_universal(binary, tunables)
+        // Compute the needed instrumentation
+        struct MaxStackCfg;
+        impl finite_wasm::max_stack::Config for MaxStackCfg {
+            fn size_of_value(&self, _ty: finite_wasm::wasmparser::ValType) -> u8 {
+                1
+            }
+            fn size_of_function_activation(
+                &self,
+                locals: &prefix_sum_vec::PrefixSumVec<finite_wasm::wasmparser::ValType, u32>,
+            ) -> u64 {
+                // Number of locals plus 1 for function metadata
+                u64::try_from(locals.max_index().map(|l| l.saturating_add(2)).unwrap_or(1)).unwrap()
+            }
+        }
+        struct GasCfg<'a>(&'a dyn Tunables);
+        macro_rules! gas_cost {
+            (@@mvp $_op:ident $_self:ident $({ $($_arg:ident: $_argty:ty),* })? => visit_block) => {
+                0
+            };
+            (@@mvp $_op:ident $_self:ident $({ $($_arg:ident: $_argty:ty),* })? => visit_end) => {
```

Comment:
Drive-by comment: you will want same treatment for `visit_else` as it acts pretty much just like `visit_end` for the purposes of the gas analysis.